api/v1alpha1/nodepool_types.go

```diff
@@ -790,6 +817,20 @@ type NodePoolCondition struct {
 	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
 }
 
+// NodePoolPlatformStatus contains specific platform statuses
+type NodePoolPlatformStatus struct {
+	// KubeVirt contains the KubeVirt platform statuses
+	// +optional
+	KubeVirt *KubeVirtNodePoolStatus `json:"kubeVirt,omitempty"`
+}
+
+// KubeVirtNodePoolStatus contains the KubeVirt platform statuses
+type KubeVirtNodePoolStatus struct {
+	// CacheName holds the name of the cache DataVolume, if exists
+	// +optional
+	CacheName *string `json:"cacheName,omitempty"`
```

Comment:
LIST is only required when we don't have a entry here for the current release.

Status - logic would be NP
- if we have DV on status for existing release (continue)
- else perform list to see if we have it already, then record it.

Creation - logic would be in NP 1:1 relationship between NP and cached DV
- Cached DV has label/annotation with nodepool uid associated in it
- if we have DV in status for existing release, use it
- else perform LIST and see if it already exists, if so use it, and cleanOldCaches()
- else create the DV and cleanOldCaches()

Deletion during np deletion
- cleanAllCache() - new function to clean up all cached DVs associated with NP. 

Annotation ties DV to Release and NodePool